buildlib/jucx/jucx-test.yml

```diff
@@ -26,6 +26,9 @@ jobs:
       - bash: |
           set -xeE
           source buildlib/az-helpers.sh
+          if [ "${{ parameters.name }}" == "gpu" ]; then
```

Comment:
maybe it should be part of try_load_cuda_env? 
or at least do it after loading cuda env

Reply:
'try_load_cuda_env' is rather forgiving, as it's used by non-GPU tests too. 
A separate function is needed to only run for GPU tests.

Any reason to load the cuda env if there is no GPU?
Per the "fail fast" concept, a pipeline should halt whenever a crucial condition is detected. This should save resources and provide fast feedback.